manager/controllers/step_controller.go

```diff
@@ -77,7 +77,9 @@ func (r *StepReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 	if step.GetDeletionTimestamp() != nil {
 		if controllerutil.ContainsFinalizer(step, stepFinalizer) {
-			r.stopMetricsCacheLoop(step)
+			if err := r.stopMetricsCacheLoop(step); err != nil {
```

Comment:
can we make sure we do not have an unlimited pool of workers? https://gobyexample.com/worker-pools

Reply:
also, what happens on restart

Reply:
My understanding is, when restarts, all the watched object will go through reconciliation.